File: src/main/java/model/rules/roundpoints/SumOpponentsPoints.java
package model.rules.roundpoints;

import model.Move;
import model.Tile;
import model.players.Player;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

public class SumOpponentsPoints implements RoundPoints {

    @Override
    public int execute(Move move, List<Player> players, LinkedList<Tile> board) {
        List<Integer> sums = new ArrayList<>();
        players.stream().map(p -> p.getTiles()).forEach(tiles -> sums.add(tiles.stream().mapToInt(t -> t.sum()).sum()));
        return sums.stream().reduce(0, (a, b) -> a + b);
    }    
}

File: docs/scripts/model/rules/roundpoints/SumOpponentsPoints.js
export default class SumOpponentsPoints {

    execute(move, players, board) {
        let sumF = (a, b) => a + b;
        let sums = [];
        players.map(p => p.getTiles()).forEach(tiles => tiles.length === 0 ? 0 : sums.push(tiles.map(t => t.sum()).reduce(sumF, 0)));
        return sums.reduce(sumF, 0);
    }    
}
